bootstrap/eks/config/manager/manager.yaml

```diff
@@ -25,5 +25,18 @@ spec:
         name: manager
       terminationGracePeriodSeconds: 10
       tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/control-plane
         - effect: NoSchedule
           key: node-role.kubernetes.io/master
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-role.kubernetes.io/control-plane
+                operator: Exists
+            # remove once usage of node-role.kubernetes.io/master is removed from Kubernetes
+            - matchExpressions:
+              - key: node-role.kubernetes.io/master
+                operator: Exists
```

Comment:
Given these expressions will be AND'd, then this won't work on older versions of Kubernetes that don't set control-plane, and will also stop working when `master` is removed.

@neolit123 , was there any guidance for appdevs on how best to handle this?

Reply:
we do not have a guide for node affinity and we assumed that k8s constructs are flexible enough to manage this, are well documented, and users will know how to use them.

you can ask someone to amend the KEP if you think it's worthy:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint

https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity

> If you specify multiple nodeSelectorTerms associated with nodeAffinity types, then the pod can be scheduled onto a node if one of the nodeSelectorTerms can be satisfied.

> If you specify multiple matchExpressions associated with nodeSelectorTerms, then the pod can be scheduled onto a node only if all matchExpressions is satisfied.

i'm reading this as - one of the nodeSelectorTerms is a sufficient match, but if a nodeSelectorTerms has multiple matchExpressions they must all match within it.

Reply:
Ah yes, multiple nodeSelectorTerms will make it work. Cool, I think that's fine then.

Thanks for that!